internal/castai/castai_test.go

```diff
@@ -89,3 +94,70 @@ func TestClient_SendDelta(t *testing.T) {
 
 	require.NoError(t, err)
 }
+
+func TestCreateTLSConfig(t *testing.T) {
+	t.Run("should populate tls.Config RootCAs when valid certificate presented", func(t *testing.T) {
+		r := require.New(t)
+
+		t.Cleanup(config.Reset)
+		t.Cleanup(os.Clearenv)
+
+		r.NoError(os.Setenv("TLS_CA_CERT_FILE", string(`
+-----BEGIN CERTIFICATE-----
+MIIC/zCCAeegAwIBAgIUHD6uJtqPjkKWI0OyZ3ZSyXJt5LUwDQYJKoZIhvcNAQEL
+BQAwDzENMAsGA1UECgwEVGVzdDAeFw0yMzA5MDgwODMyMTNaFw0yNDA5MDcwODMy
+MTNaMA8xDTALBgNVBAoMBFRlc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK
+AoIBAQD5PhKa0D8VfVai7cZhDpBdFLHcEaYlYtgm3yyDMe1jxWn7SYl5GbC7+8+o
+C2hZwYVXqik65BAg6QSf0waBy98c4uuD9yw04mLxRtdnsAxdXIewO/i0+zUqLayR
+HXTTU9LEAENu8Kk27ms8nzlx9jzM/2cNgjmXmksyZrDkdeGcs2uwWV8ZwcAcW+aV
+CNHow/WG+3H5pD4rI0DLahUiph+CHDT7YZchyjhNVq756UfaJ57sTRmS13uwOTQ4
+ZdCxKgO656yManYkm2k9XaIT+vqIy0m8OxkWMulsTy0WfwmBCCNMUICdr95ppycl
++dkHyqyvty8VeB0kteVh+j3TX4MXAgMBAAGjUzBRMB0GA1UdDgQWBBQVK59ct3yY
+2/sjwFGwoP9+eV9MyzAfBgNVHSMEGDAWgBQVK59ct3yY2/sjwFGwoP9+eV9MyzAP
+BgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQCjCuV2urgJtZWvX1df
+5FGh4o3jAC8dEQZMJvW9C1/gPlDGs/z9/2Fo74hYDW87hJnayPFaveRDYhHul6sN
+J1sUKDdDUkVIarRS5/G39xNCvjYFRuag2vqO6PzxEe0h8NtVrCxkvG1qrgV2y0Tx
+0mZiX8xgjAocjdZV4kPaVL+b5+N8LiR/yw1WQLPNvuZF4hsLqB0uaDT5ekFRnUM1
+yxZ0LjlYV+vtoEgA9tLwVGR5Tu30ggVMuZ4ZwIu6XMR3QCNHz7HjKFVmG3t8RgjI
+OiGvAsXa3skJBqeNXUXQR/GsRm7UfJV9gVt2sJvmeHPx1PfQytWW8KFJJL2Qe+7p
+rWFE
+-----END CERTIFICATE-----
```

Comment:
The cert is set to expire at Sep  7 08:32:13 2024 which means the test will start failing one year later. Set the expiration to some insane future date.

Reply:
Updated cert to one valid until 2215.